packages/clerk-js/src/ui/components/SignIn/SignIn.tsx

```diff
@@ -170,6 +187,13 @@ function SignInRoot() {
    */
   usePreloadSignUp(signInContext.isCombinedFlow);
 
+  // `experimental.withSessionTasks` will be removed soon in favor of checking via environment response
+  usePreloadSessionTask(signInContext.withSessionTasks);
+
+  React.useEffect(() => {
+    return __internal_setComponentNavigationContext?.({ basePath, navigate });
+  }, []);
```

Comment:
consider adding `basePath` and `navigate` to the dependency array
```suggestion
  React.useEffect(() => {
    return __internal_setComponentNavigationContext?.({ basePath, navigate });
  }, [basePath, navigate]);
```